apps/rush-lib/src/logic/Git.ts

```diff
@@ -104,6 +112,37 @@ export class Git {
     return emailResult.result;
   }
 
+  /**
+   * Get the folder where Git hooks should go for the current working tree.
+   * Returns undefined if the current path is not under a Git working tree.
+   */
+  public static getHooksFolder(): string | undefined {
+    const repoInfo: gitInfo.GitRepoInfo | undefined = Git.getGitInfo();
+    if (repoInfo && repoInfo.worktreeGitDir) {
+      return path.join(repoInfo.worktreeGitDir, 'hooks');
+    }
+    return undefined;
+  }
+
+  /**
+   * Get information about the current Git working tree.
+   * Returns undefined if the current path is not under a Git working tree.
+   */
+  public static getGitInfo(): Readonly<gitInfo.GitRepoInfo> | undefined {
+    if (!Git._checkedGitInfo) {
+      try {
+        const repoInfo: gitInfo.GitRepoInfo = gitInfo();
+        if (Git.isPathUnderGitWorkingTree(repoInfo)) {
+          Git._gitInfo = repoInfo;
+        }
+      } catch (ex) {
+        // ignore
+      }
```

Comment:
What kind of exceptions are we expecting to catch and ignore here?

If this throws and catches an exception as part of a normal expected state that is not an error (an "unexceptional exception"), is there a test we can add that would avoid that?  (Unexceptional exceptions make debugging harder, because when you try to use the "break on exceptions" feature it constantly stops at exceptions that don't reflect any actual problem.)

Reply:
Added a comment and changed the try/catch to wrap only the most relevant part (which actually is very unlikely to throw)